src/uct/ib/ud/base/ud_ep.c

```diff
@@ -672,9 +676,11 @@ uct_ud_ep_process_ack(uct_ud_iface_t *iface, uct_ud_ep_t *ep,
     }
 
     ep->tx.acked_psn = ack_psn;
-    ucs_assertv(UCT_UD_PSN_COMPARE(ep->tx.acked_psn, <, ep->tx.psn),
-                "ep %p: acked_psn=%u must be smaller than current_psn=%u", ep,
-                ep->tx.acked_psn, ep->tx.psn);
+    ucs_assertv(UCT_UD_PSN_COMPARE(ep->tx.acked_psn, <, ep->tx.psn) ||
```

Comment:
maybe better drop all RX after we did flush(CANCEL) on the ep?

Reply:
for RC/DC, we moved QP to error state, which would drop all RX..
if drop all rx, maybe we could combine it with existing branch to reduce overhead

Reply:
yes, I know and now I see that makes sense to align with other IB transport's logic